### DISCUSSION TITLE
How use the "transient prompt" feature in Starship, if possible?

### DISCUSSION THREAD

**Discussion by EdmundsEcho** [starship/starship#3540] [Q&A]
Great utility.  Thank you.

I have the two-line prompt.  The first line shows the user, git status, package version etc... The second line has the prompt.

When I execute a command, both lines become part of the "visual log", the display that tracks the record of previously executed commands.  

The issue: the user, git status, package information etc...  line is repeatedly printed. I would like to clear that away before the output is added to the "visual log".  This way I only see the command and the command output - for what has been executed.

The current prompt would continue to show the two-line version.

How would I go about making this possible?

**Comment by chipbuster** [comment]
It sounds like you're trying to implement something like ["transient prompt" from pl10k](https://github.com/romkatv/powerlevel10k#transient-prompt) (but without the command history)?

Right now, this is *incredibly* hard to do. You need a shell with pretty advanced features (I'd only even think about doing it with zsh or fish), and very complex integration scripts (to the point where pl10k's implementation can straight-up segfault zsh).

If you'd like to take a stab at it, the creator of pl10k left some pointers over on #888.

**Reply by EdmundsEcho** [reply]
Exactly like transient prompt.  We keep the command and output in the history.  We delete the git status and other information before appending to the terminal history. I'll take a look at the pointers.  

Based on your response the function call before rendering, what seems to be a feature of Starship, does not have the scope do so.  Correct?

I'll take a look at the pointers.

**Comment by chipbuster** [comment]
Oh...then again, it occurs to me that if you 

 - don't need the command history and
 - are willing to stick to ANSI terminals only

then it might be possible to implement this by getting the shell to emit a "clear screen" control code. This will make your scrollback history long and annoying to scroll through, but it would only require a little bit of extra code.

**Comment by EdmundsEcho** [comment]
I'm using fishshell.  The "transient prompt" feature seems to be something to expect in v3.4 of fishshell ([#7602](https://github.com/fish-shell/fish-shell/issues/7602), [#85](https://github.com/IlanCosman/tide/issues/85)).  When it's out I'll be sure to play with it to see how to get it work with Starship.
